lib/commands/register.js

```diff
@@ -110,16 +111,28 @@ function confirm(properties) {
                     default: plugin.toString() || 'not specified',
                     required: true
                 },
-                repository: {
-                    description: chalk.cyan('repository'),
-                    pattern: /([A-Za-z0-9]+@|http(|s)|git\:\/\/)([A-Za-z0-9.]+)(:|\/)([A-Za-z0-9\-\.\/]+)(\.git)?/,
+                repositoryType: {
+                    description: chalk.cyan('repository type (e.g. git)'),
+                    message: 'Please provide a repository type',
+                    pattern: /.+/,
                     type: 'string',
-                    default: properties.repository || 'not specified',
+                    default: properties.repository ? properties.repository.type : undefined,
```

Comment:
Minor but is it worth defaulting to 'git' as it will be the most common and mean the user can just hit return again?

Reply:
Actually, further to that, since git is required according to the regex below can this user entry be removed and added automatically?

Reply:
Yeah I wasn't clear on what was allowed for type/url. The spec isn't very explanatory: https://github.com/bower/spec/blob/master/json.md#repository. Shall we assume it must be of type `git` and a `git://` protocol? Now that I think on, the underlying bower register command fails if you don't have a `git://` protocol. So, drop asking the user for a type and amend the regex to only allow `git://` protocol; ok with you?

Reply:
Yup, sounds good to me